pkg/sanity/snapshot.go

```diff
@@ -0,0 +1,317 @@
+			snapID           string
+		)
+
+		BeforeEach(func() {
+			var err error
+			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, make(map[string]string))
+			numVolumesBefore = len(volumes)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			var err error
+
+			err = volumedriver.Delete(volumeID)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = volumedriver.Delete(snapID)
+			Expect(err).ToNot(HaveOccurred())
+
+			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, nil)
+			Expect(err).ToNot(HaveOccurred())
+			numVolumesAfter = len(volumes)
+		})
+
+		It("Should create Volume successfully for snapshot", func() {
+
+			By("Creating the volume")
+
+			var err error
+			var size = 3
+			vr := &api.VolumeCreateRequest{
+				Locator: &api.VolumeLocator{
+					Name: "create-for-snapshot",
+					VolumeLabels: map[string]string{
+						"class": "create-class",
+					},
+				},
+				Source: &api.Source{},
+				Spec: &api.VolumeSpec{
+					Size:      uint64(size * GIGABYTE),
+					HaLevel:   1,
+					Format:    api.FSType_FS_TYPE_EXT4,
+					BlockSize: int64(512 * KILOBYTE),
+					Cos:       api.CosType_LOW,
+				},
+			}
+
+			volumeID, err = volumedriver.Create(vr.GetLocator(), vr.GetSource(), vr.GetSpec())
+
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking if volume created successfully with the provided params")
+			testIfVolumeCreatedSuccessfully(volumedriver, volumeID, numVolumesBefore, vr)
+
+			By("Creating a snapshot based on the created volume")
+
+			loc := &api.VolumeLocator{
+				Name: "snapshot-of-" + volumeID,
+			}
+
+			snapID, err = volumedriver.Snapshot(volumeID, true, loc)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(snapID).To(Not(BeNil()))
+
+			By("Checking the Parent field of the created snapshot")
+
+			volumes, err := volumedriver.Inspect([]string{loc.GetName()})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(volumes).NotTo(BeEmpty())
+
+			Expect(volumes[0].GetSource().GetParent()).To(BeEquivalentTo(volumeID))
+			Expect(volumes[0].GetReadonly()).To(BeTrue())
+		})
+	})
+
+	Describe("Volume Snapshot Enumerate", func() {
+
+		var (
+			numVolumesBefore int
+			numVolumesAfter  int
+			volumeID         string
+			snapID           string
+			snapIDs          []string
+		)
+
+		BeforeEach(func() {
+			var err error
+			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, make(map[string]string))
+			numVolumesBefore = len(volumes)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			var err error
+
+			err = volumedriver.Delete(volumeID)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, snapID := range snapIDs {
+				err = volumedriver.Delete(snapID)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			volumes, err := volumedriver.Enumerate(&api.VolumeLocator{}, nil)
+			Expect(err).ToNot(HaveOccurred())
+			numVolumesAfter = len(volumes)
+		})
+
+		It("Should enumerate Volume snapshots", func() {
+
+			By("Creating the volume")
+
+			var err error
```

Comment:
nit: create the volume once in BeforeEach of the snapshot test suite. so that you don't have to create it in every "It"

Reply:
Every Describe block is a new test , so sometimes it could be that there is a different configuration or different number of volumes are to be created